File: JH/DAY5/11053.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
int arr[1001];
int dp[1001];

int main(){

    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    cin >> n;

    for (int i = 0; i <n ; ++i) {
        cin >> arr[i];
        dp[i] =1;
    }

    int ans = 1;
    for (int i = 0; i <n ; ++i) {
        for (int j = 0; j <i ; ++j) {
            if(arr[i] > arr[j]) {
                dp[i] = max(dp[i],dp[j]+1);
                ans = max(ans,dp[i]);
                //cout << dp[i] << "\n";
            }
        }
        //cout << "\n";
    }

    cout << ans << "\n";






}

File: AJM/DAY6/11053.py
#가장 긴 증가하는 부분 수열

"""
0  1  2  3  4  5
10 20 10 30 20 50

dp
1  1  1  1  1  1
1  2
1  2  1
1  2  1  3
"""
n = int(input())
arr = list(map(int,input().split()))

dp = [1]*n

for i in range(n):
    for j in range(i):
        if arr[i] > arr[j]:
            dp[i] = max(dp[i],dp[j]+1)
    
print(max(dp))
